auth/store_test.go

```diff
@@ -0,0 +1,192 @@
+		if err != ErrUserAlreadyExist {
+			t.Fatalf("expected %v, got %v", ErrUserAlreadyExist, err)
+		}
+	} else {
+		t.Fatalf("expected %v, got %v", ErrUserAlreadyExist, err)
+	}
+}
+
+func TestAuthenticate(t *testing.T) {
+	b, tPath := backend.NewDefaultTmpBackend()
+	defer func() {
+		b.Close()
+		os.Remove(tPath)
+	}()
+
+	as := NewAuthStore(b)
+
+	// auth a non-existing user
+	_, err := as.Authenticate("foo", "bar")
+	if err != nil {
+		if !strings.Contains(err.Error(), "auth: authentication failed, invalid user ID or password") {
+			t.Fatalf("expected 'auth: authentication failed, invalid user ID or password' error, got %v", err)
+		}
+	} else {
+		t.Fatalf("expected 'invalid password...' error, got %v", err)
+	}
+
+	// auth an existing user with correct password
+	ua := &pb.AuthUserAddRequest{Name: "foo", Password: "bar"}
+	_, err = as.UserAdd(ua)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = as.Authenticate("foo", "bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// auth an existing user but with wrong password
+	_, err = as.Authenticate("foo", "")
+	if err != nil {
+		if !strings.Contains(err.Error(), "auth: authentication failed, invalid user ID or password") {
+			t.Fatalf("expected 'auth: authentication failed, invalid user ID or password' error, got %v", err)
+		}
+	} else {
+		t.Fatalf("expected 'invalid password...' error, got %v", err)
+	}
+}
+
+func TestUserDelete(t *testing.T) {
+	b, tPath := backend.NewDefaultTmpBackend()
+	defer func() {
+		b.Close()
+		os.Remove(tPath)
+	}()
+
+	as := NewAuthStore(b)
+
+	ua := &pb.AuthUserAddRequest{Name: "foo"}
+	_, err := as.UserAdd(ua)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// delete an existing user
+	ud := &pb.AuthUserDeleteRequest{Name: "foo"}
+	_, err = as.UserDelete(ud)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// delete a non-existing user
+	_, err = as.UserDelete(ud)
+	if err != nil {
+		if err != ErrUserNotFound {
+			t.Fatalf("expected %v, got %v", ErrUserNotFound, err)
+		}
+	} else {
+		t.Fatalf("expected %v, got %v", ErrUserNotFound, err)
+	}
+}
+
+func TestUserChangePassword(t *testing.T) {
+	b, tPath := backend.NewDefaultTmpBackend()
+	defer func() {
+		b.Close()
+		os.Remove(tPath)
+	}()
+
+	as := NewAuthStore(b)
+
+	_, err := as.UserAdd(&pb.AuthUserAddRequest{Name: "foo"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = as.Authenticate("foo", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = as.UserChangePassword(&pb.AuthUserChangePasswordRequest{Name: "foo", Password: "bar"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = as.Authenticate("foo", "bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// change a non-existing user
+	_, err = as.UserChangePassword(&pb.AuthUserChangePasswordRequest{Name: "foo-test", Password: "bar"})
+	if err != nil {
+		if err != ErrUserNotFound {
+			t.Fatalf("expected %v, got %v", ErrUserNotFound, err)
+		}
+	} else {
+		t.Fatalf("expected %v, got %v", ErrUserNotFound, err)
+	}
+}
+
+func TestRoleAddGrant(t *testing.T) {
```

Comment:
This test tests too many stuff. we should not test permission grant in this test.